### DISCUSSION TITLE
Get exact informations from document + write

### DISCUSSION THREAD

**Discussion by Tomp0uce** [FlowiseAI/Flowise#3161] [Q&A]
Hi guys, I have a small questions for you : 
I try to use flowise for its RAG abilities : Indeed I want to provide to the workflow one big txt file containing more than 300 requirements for a technical device. The txt document is structured as follow : 

"Requirement #1:
Req. ID : XXX_YYY_001
Req doc: Requirement001.docx
Description: The device should be used for temperature between 0 and 60°C
Type of Evidence: Test
Test ID: Environment"

I use the following workflow at the moment: 
![image](https://github.com/user-attachments/assets/72f37767-f996-4abb-bf9f-81bba6800cd4)

If I ask really specific question on a unique requirement sometimes the answer is ok, sometimes not (sometimes the bot don't even find an answer to questions with the exact same wording than in the txt file). When I ask question that should get their answer from multiple requirements then the answer is even catastrophic.

for exemple :
- If I ask "which standard do the device need to follow?" I got a partial list of the standard listed but it's not complete. 

-  If I ask "how many requirement there is for this device", the answer provided is :
"There are 4 documents (id '0', id '1', id '2', and id '3') with a total of 31 requirements listed within them.
Here's the breakdown:
Document id '0' has 4 requirements.
Document id '1' has 3 requirements.
Document id '2' has 3 requirements.
Document id '3' has 2 requirements."

The number or requirement per snippets is good but It lacks a lot off requirements in total (>300 in the file) and the sum is not even good.

- If I ask "list me all requirement description of requirements with test id as environment" the answer is focus on 2 or 3 snippets which is clearly insufficient because I have more than 20 requirements with "environment" as a test ID.

Do you have any clues on how to get better results for those questions? (beware one constraint is to have only block that are fully private (no access to online services)

Second steps of my tests are to try to get the workflow to use  another document as a template  to write a test protocol to qualify some of thoses requirement through tests. 
Due to the fact that the Bot don't seems to get good knowledge of the whole document the test procedure are really bad and don't follow well the template. Do you have any advice on how to perform this task? Use of agent?

Thanks in advance for your help!

**Comment by alexandregcloficial** [comment]
I have the same problem when searching and responding to data. When I attach, for example, the same file in the GPT chat, it returns the correct information, but in the Flowise agent, it does not.
